src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/Main.java

```diff
@@ -43,28 +44,34 @@ public static void main(String[] args) throws IOException {
         final String[] jarFilesArray = jarFiles.split(",");
 
         final File outputDir = new File(args[1]);
-        if (!outputDir.exists()) {
-            if (!outputDir.mkdirs()) {
-                System.out.printf("Failed to create output directory %s%n", outputDir);
-            }
-        }
 
         System.out.println("Running with following configuration:");
         System.out.printf("  Output directory: '%s'%n", outputDir);
 
-        for (final String jarFile : jarFilesArray) {
-            System.out.printf("  Processing input .jar file: '%s'%n", jarFile);
+        Arrays.stream(jarFilesArray).forEach(jarFile -> run(new File(jarFile), outputDir));
+    }
 
-            final File file = new File(jarFile);
-            if (!file.exists()) {
-                System.out.printf("Cannot find file '%s'%n", file);
-                System.exit(-1);
-            }
+    /**
+     * Runs APIView parser and returns the output file path.
+     */
+    public static File run(File jarFile, File outputDir) {
+        System.out.printf("  Processing input .jar file: '%s'%n", jarFile);
 
-            final String jsonFileName = file.getName().substring(0, file.getName().length() - 4) + ".json";
-            final File outputFile = new File(outputDir, jsonFileName);
-            processFile(file, outputFile);
+        if (!jarFile.exists()) {
+            System.out.printf("Cannot find file '%s'%n", jarFile);
+            System.exit(-1);
         }
+
+        if (!outputDir.exists()) {
+            if (!outputDir.mkdirs()) {
+                System.out.printf("Failed to create output directory %s%n", outputDir);
```

Comment:
Should we `System.exit(-1)` from here if output directory cannot be created?

Reply:
Done